### PR TITLE
PS-1798 processing processed tags process

### PR DESCRIPTION
**Pull request by pivnicek**
good morning @odinuv, here's a brain walloper for breakfast :-).  please let me know if there are some cases I've overlooked in the test, I'm sure there are.

### CODE REVIEW

**Comment by odinuv** [comment]
Looks a bit  😄

File: src/Helper/TagsRewriteHelper.php [hunk 1]
```diff
@@ -40,21 +41,44 @@ public static function rewriteFileTags(
 
         if (!empty($fileConfiguration['source']['tags'])) {
             $oldTagsList = $fileConfiguration['source']['tags'];
-            $newTagsList = self::overwriteSourceTags($prefix, $oldTagsList);
+            $includeTags = array_filter($oldTagsList, function ($tag) {
```
Comment (odinuv) [inline src/Helper/TagsRewriteHelper.php:44]: please add a comment with link to the confluence doc here so that we know what this beast is for

**Review by odinuv — COMMENTED** [review]
Great, i think it does exactly what it's supposed to do. Not sure why this https://dev.azure.com/keboola-dev/input-mapping/_build/results?buildId=7292&view=logs&j=c634f45f-4c77-54b1-457c-e7837386f5d5&t=dedcc033-d22f-59bc-3156-d53637b22762&l=904 doesn't pass though 

Also It would be nice to add a couple of individual tests on the TagsRewriteHelperTest

File: src/Helper/TagsRewriteHelper.php [hunk 2]
```diff
@@ -67,7 +67,7 @@ public static function rewriteFileTags(
                 $excludeTags = $newExcludeTags;
             }
 
-            if (self::hasFilesWithSourceTags($clientWrapper, $includeTags)) {
+            if (self::hasFilesWithSourceTags($clientWrapper, $newIncludeTags)) {
```
Comment (pivnicek) [inline src/Helper/TagsRewriteHelper.php]: @odinuv FYI that test that failed previously caught this very important mistake! :-)

Comment (odinuv) [reply src/Helper/TagsRewriteHelper.php]: i seriously love when this happens because i completely missed that

**Review by odinuv — APPROVED** [review]

File: src/Helper/TagsRewriteHelper.php (hunk 2 above)
Comment (pivnicek) [reply src/Helper/TagsRewriteHelper.php]: me too :-)

**Pull request merged**